lib/stat_tracker.rb

```diff
@@ -117,143 +116,77 @@ def find_team_by_id(id)
-      avg_score
-    end.first
+  # consider combining above two methods taking an addtl arg for max_by or min_by
 
-    find_team_by_id(highest_scoring_visitor_id).team_name
+  def best_offense
+    team_avgs = team_averages(sort_scores_by_team(@game_teams))
+    find_team_by_id(team_with_highest_average_score(team_avgs)).team_name
   end
 
-  def highest_scoring_home_team
-    home_teams = @game_teams.find_all do |game_team|
-      game_team.hoa == "home"
-    end
-
-    sorted_home_teams = home_teams.reduce({}) do |team_scores, game|
-      if team_scores[game.team_id].nil?
-        team_scores[game.team_id] = [game.goals.to_i]
-      else
-        team_scores[game.team_id] << game.goals.to_i
-      end
-      team_scores
-    end
+  def worst_offense
+    team_avgs = team_averages(sort_scores_by_team(@game_teams))
+    find_team_by_id(team_with_lowest_average_score(team_avgs)).team_name
+  end
 
-    avgs_by_team = {}
-    sorted_home_teams.each do |home_team_id, scores_array|
-      avgs_by_team[home_team_id] = (scores_array.sum / scores_array.count.to_f)
+  def find_game_teams(home_or_away)
+    @game_teams.find_all do |game_team|
+      game_team.hoa == home_or_away
     end
+  end
 
-    highest_scoring_home_id = avgs_by_team.max_by do |_visiting_team_id, avg_score|
-      avg_score
-    end.first
+  def highest_scoring_visitor
+    away_teams = find_game_teams("away")
+    sorted_away_team_scores = sort_scores_by_team(away_teams)
+    team_avgs = team_averages(sorted_away_team_scores)
+    find_team_by_id(team_with_highest_average_score(team_avgs)).team_name
+  end
 
-    find_team_by_id(highest_scoring_home_id).team_name
+  def highest_scoring_home_team
+    home_teams = find_game_teams("home")
+    sorted_home_team_scores = sort_scores_by_team(home_teams)
+    team_avgs = team_averages(sorted_home_team_scores)
+    find_team_by_id(team_with_highest_average_score(team_avgs)).team_name
   end
 
   def lowest_scoring_visitor
-    away_teams = @game_teams.find_all do |game_team|
-      game_team.hoa == "away"
-    end
-
-    sorted_away_teams = away_teams.reduce({}) do |team_scores, game|
-      if team_scores[game.team_id].nil?
-        team_scores[game.team_id] = [game.goals.to_i]
-      else
-        team_scores[game.team_id] << game.goals.to_i
-      end
-      team_scores
-    end
-
-    avgs_by_team = {}
-    sorted_away_teams.each do |visiting_team_id, scores_array|
-      avgs_by_team[visiting_team_id] = (scores_array.sum / scores_array.count.to_f)
-    end
-
-    lowest_scoring_visitor_id = avgs_by_team.min_by do |_visiting_team_id, avg_score|
-      avg_score
-    end.first
-
-    find_team_by_id(lowest_scoring_visitor_id).team_name
+    away_teams = find_game_teams("away")
+    sorted_away_team_scores = sort_scores_by_team(away_teams)
+    team_avgs = team_averages(sorted_away_team_scores)
+    find_team_by_id(team_with_lowest_average_score(team_avgs)).team_name
```

Comment:
@rrabinovitch I can't tell if this is indented properly. In the review it looks like it's outside of `def lowest_scoring_visitor`